src/core/mount.c

```diff
@@ -487,7 +487,7 @@ static int mount_add_default_ordering_dependencies(Mount *m, MountParameters *p,
                  * it's not technically part of the basic initrd filesystem itself, and so
                  * shouldn't inherit the default Before=local-fs.target dependency. */
```

Comment:
If the change is approved, then please update the comment.

Reply:
That is a good point. This comment was added in 83cdc870949823b5b9fa04dd76e952d42faab0b1 which said:

   But it had the bad side effect to also remove all other default dependencies as
   well. Thus if an initrd mount was using _netdev, the network dependencies were
   missing.
   
   In general fstab-generator shouldn't use DefaultDependecies=no because it can
   handle only a small set of the default dependencies the rest are dealt by pid1.

So it indeed sounds like the sysroot mounts must not be part of the `local-fs.target`, and that whe should add a separate synchronization target. `initrd-sysroot-pre.target`? And then order `systemd-hibernate-resume@.service` before it, and the mount+fsck stuff after.

Reply:
Hmm, though looking at this again, we don't really need a separate target. We *could* add it, but it seems that we can do just fine without it. So I think we would only need to adjust the comment a bit to explain why we use `After=local-fs-pre.target` but not `Before=local-fs.target`.

Reply:
> Hmm, though looking at this again, we don't really need a separate target. We _could_ add it, but it seems that we can do just fine without it.

I cannot follow that. Could you explain why?